utils/src/tree/AzExtTreeDataProvider.ts

```diff
@@ -61,6 +61,10 @@ export class AzExtTreeDataProvider implements IAzExtTreeDataProviderInternal, ty
             ti.tooltip = await treeItem.resolveTooltip();
         }
 
+        if (treeItem.resolve && treeItem.resolveResult === undefined) {
+            await treeItem.resolve(false);
```

Comment:
This approach isn't going to do what we want. Resolving a tree item (in the sense of "resolve" that we created, not the VSCode sense) needs to be able to change the label, description, etc. In the spec for VSCode's `resolveTreeItem`, it states that the tree items are already shown in the UI (so label, description, etc. can't be changed), _and_ also says that `onDidChangeTreeData` should _not_ be triggered (which eliminates it as a workaround for the lack of label etc. change).